code/modules/projectiles/guns/projectile/shotgun.dm

```diff
@@ -143,6 +149,71 @@
 			if(istype(user, /mob/living/carbon/human) && src.loc == user)
 				var/mob/living/carbon/human/H = user
 				H.update_inv_hands()
+	if(istype(A, /obj/item/stack/rods))
+		to_chat(user, "<span class='notice'>You start jamming a rod into \the [src]'s barrels to try and loosen the shells...</span>")
+		if(do_after(user, src, 60))
+			jammed = 0
+			to_chat(user, "<span class='notice'>You manage to loosen the shells.</span>")
+
+/obj/item/weapon/gun/projectile/shotgun/doublebarrel/verb/toggle_doubleshot()
+	set name = "Toggle Shooting Both Barrels"
+	set category = "Object"
+	if((world.time >= last_fired + fire_delay) && !doubleshooting)
+		doubleshot = !doubleshot
+		if(!doubleshot)
+			fire_delay = initial(fire_delay)
+		to_chat(usr, "You switch \the [src]'s fire selector to [doubleshot ? "fire both barrels at once" : "fire one barrel at a time"].")
+
+/obj/item/weapon/gun/projectile/shotgun/doublebarrel/Fire(atom/target as mob|obj|turf|area, mob/living/user as mob|obj, params, reflex = 0, struggle = 0)
+	if(doubleshot && (getLiveAmmo() == 2)) //ANGERY
+		var/atom/reverse = locate(2*user.x - target.x, 2*user.y - target.y, target.z)
+
+		if(ready_to_fire())
+			fire_delay = 0
+		else
+			return
+		if(defective && prob(5))
+			to_chat(user, "<span class='danger'>\The [src] can't handle the pressure of firing two shells at once!</span>")
+			explosion(get_turf(loc), -1, 0, 2)
+			user.drop_item(src, force_drop = 1)
+			qdel(src)
+		else
+			doubleshooting = 1
+			recoil = 2 * initial(recoil)
+			..()
+			..()
+			message_admins("[usr] just fired both barrels out of \his [src].")
+			fire_delay = 20
+			recoil = initial(recoil)
+			doubleshooting = 0
+			if(prob(50)) // Total chance to fuck up
+				var/mob/living/carbon/human/H = user
+				switch(pick(1,2,3,4))
```

Comment:
You could probably tell what those numbers actually do or use defines.
```dm
#define JAMMED 1
#define RECOIL 2
#define KNOCKDOWN 3
#define BREAK_ARM 4
var/severity = pick(JAMMED, ...)
switch(severity)
         if (JAMMED)
               // ...
```
But that's not a major concern, just a little bit of readability

Reply:
Well, the recoil effect selection was concentrated in only a dozen lines so an attentive reader could've figured it out easily, especially when reading the comments I added to the code.
More readability can't hurt though, so I did it anyway.